### PR TITLE
Feature/pvt events v.0.1.0

### PR DESCRIPTION
**Pull request by ricardo123321**
We created a private events page that allows us to make and attend events that belong to different users and we establish numerous methods of validations so the information provided will be one acceptable, and those who are not signed can see the events, but those who sign can attend and leave the events and see who made them and the upcoming or past events

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+  # POST /events
+  # POST /events.json
+  def create
+    @event = current_user.events.build(event_params)
+    # User.find_by(name: session[:current_user_name])
+    # @current_user = User.where(name: session[:current_user_name])
+    # @current_user1 = @current_user.name
+
+    # @event = @current_user1.events.build(event_params)
```
Comment (kobiyoyo) [inline app/controllers/events_controller.rb]: Comments: It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history.

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    respond_to do |format|
+      if @event.save
+        format.html { redirect_to @event, notice: 'Event was successfully created.' }
+        format.json { render :show, status: :created, location: @event }
```
Comment (kobiyoyo) [inline app/controllers/events_controller.rb]: - [x] You don't need this rendered in JSON ,let remove this.

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  # PATCH/PUT /events/1
+  # PATCH/PUT /events/1.json
+  def update
+    respond_to do |format|
+      if @event.update(event_params)
+        format.html { redirect_to @event, notice: 'Event was successfully updated.' }
+        format.json { render :show, status: :ok, location: @event }
+      else
+        format.html { render :edit }
+        format.json { render json: @event.errors, status: :unprocessable_entity }
```
Comment (kobiyoyo) [inline app/controllers/events_controller.rb]: - [x] You don't need this rendered in JSON ,let remove this.

File: app/controllers/events_controller.rb [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    # @event = @current_user1.events.build(event_params)
+
+    respond_to do |format|
+      if @event.save
+        format.html { redirect_to @event, notice: 'Event was successfully created.' }
+        format.json { render :show, status: :created, location: @event }
+      else
+        format.html { render :new }
+        format.json { render json: @event.errors, status: :unprocessable_entity }
```
Comment (kobiyoyo) [inline app/controllers/events_controller.rb]: - [x] You don't need this rendered in JSON ,let remove this.

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  def show
+    @events_made_by_current_user = @user.events
+    @upcoming_events = @user.attended_events.upcoming_events
+    @prev_events = @user.attended_events.previous_events
+    # current_user1 = User.find_by_name(session[:current_user_name])
+    # current_user = current_user1.name
+    # #@events_by_user = current_user.events_by_user
+    # @events_by_user = current_user.events
+    @user = User.find(params[:id])
+  end
```
Comment (kobiyoyo) [inline app/controllers/users_controller.rb]: ###### `show` action

* [x] lists events created by user, i.e. `@created_events = current_user.events` 
* [x] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` 
* [x] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming`

File: db/schema.rb [hunk 6]
```diff
@@ -1,18 +1,41 @@
...
+    t.datetime "date"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.integer "creator_id"
+    t.index ["creator_id"], name: "index_events_on_creator_id"
+  end
+
+  create_table "invitations", force: :cascade do |t|
+    t.integer "attendee_id"
+    t.integer "attended_event_id"
```
Comment (kobiyoyo) [inline db/schema.rb:30]: - [x] Lets index this columns

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, but there are a few more things required. 
### General
- [x] Let's fix the broken image in the readme.MD file
- [x] SInce we are not building API's lets remove all the `format.json`
- [x] There should be NO application logic in the views nor helpers (no `Post.new`, `Comment.create`, `current_user.friends.find_by(...)`...). Everything connected with data processing should be encapsulated in models and used in controllers.
- [x] In the user's show page, events need to be separated( past and upcoming events)
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: app/controllers/events_controller.rb (hunk 2 above)
Comment (simandebvu) [reply app/controllers/events_controller.rb]: Removed 😄

File: app/controllers/events_controller.rb (hunk 3 above)
Comment (simandebvu) [reply app/controllers/events_controller.rb]: Removed 😄

File: app/controllers/events_controller.rb (hunk 4 above)
Comment (simandebvu) [reply app/controllers/events_controller.rb]: Removed 😄

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (simandebvu) [reply app/controllers/events_controller.rb]: Removed 😄

File: db/schema.rb (hunk 6 above)
Comment (simandebvu) [reply db/schema.rb:30]: Resolved

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (simandebvu) [reply app/controllers/users_controller.rb]:
```
    @events_made_by_current_user = @user.events
    @upcoming_events = @user.attended_events.upcoming_events
    @prev_events = @user.attended_events.previous_events
```

These methods are already in the users controller and then referenced in the show method.

**Comment by Anna-Myzukina** [comment]
## Hi team :clap: 
### You did a great job so far, but there are a few more things required.


## [REQUIRED]
#### Tests

* [x] models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
* [x] authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
* [x] event managment features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

### Make changes and submit another code review request.

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*
*Happy coding * <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

**Review by stratospherique — APPROVED** [review]
### Great work implementing the suggested changes :clap: :clap: Congrats, the project is approved :heavy_check_mark: You may merge :+1: :+1: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Comment by simandebvu** [comment]
Thank you for the review and helpful feedback 😄

**Pull request merged**